lib/internal/fs/promises.js

```diff
@@ -917,6 +920,25 @@ async function readFile(path, options) {
   return handleFdClose(readFileHandle(fd, options), fd.close);
 }
 
+function _watch(filename, options = kEmptyObject) {
```

Comment:
If we want to align the implementation of the "native" implementation, we probably should make it an async iterator as well.
```suggestion
async function* _watch(filename, options = kEmptyObject) {
```

(On macOS, `node -p 'fs.promises.watch()'` returns `Object [AsyncGenerator] {}`)